### ISSUE TITLE
AWS KMS encryption not working with sso credentials

### ISSUE THREAD

**Issue by kg-jcaron** [pulumi/pulumi#15406]
### What happened?

If you use IAM Identity Center sso to manage your aws credentials, you can use the credentials to create and manage resources through pulumi, but you can't use the same credentials as an awskms secrets encryption provider.

### Example

Running `pulumi stack change-secrets-provider "awskms://alias/MY_ALIAS?region=us-west-2&awssdk=v3&profile=MY_PROFILE"`

results in the following error:

`error: open keeper awskms://alias/MY_ALIAS?region=us-west-2&awssdk=v3&profile=MY_PROFILE: unknown query parameter "profile"`

The same profile does work with the aws cli and with pulumi in general.

Running it with `awssdk=v2` seems to run into issues with not supporting the sso-session property in the aws config file, or if you switch to the legacy credential configuration for aws, it can't seem to find the cached credentials after you login to aws.

### Output of `pulumi about`

CLI          
Version      3.105.0
Go Version   go1.21.6
Go Compiler  gc

Plugins
NAME        VERSION
aws         6.20.1
cloudflare  5.19.0
nodejs      unknown
random      4.15.1

Host     
OS       darwin
Version  14.2.1
Arch     x86_64

This project is written in nodejs: executable='/Users/XXXXXX/.asdf/shims/node' version='v20.11.0'

Backend        
Name           pulumi.com

Dependencies:
NAME                VERSION
@pulumi/aws         6.20.1
@pulumi/cloudflare  5.19.0
@pulumi/pulumi      3.105.0
@pulumi/random      4.15.1
ip6                 0.2.10
typescript          5.3.3
@types/node         18.19.14

Pulumi locates its logs in /var/folders/jy/665bhyxj7nv51dsvnvf735zw0000gp/T/ by default

### Additional context

I did test this with a couple of other versions of pulumi (include 3.105.1) and they all seem to have the same issue. I found this bug from 2 weeks ago with a very similar issue that wasclosed and deployed, but I'm still seeing the issue and can't find any good workarounds for it: https://github.com/pulumi/pulumi/issues/14647

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
We do all the interfacing with aws kms through gocloud.dev, `awssdk=v3` isn't a valid option to that. It's either "v2" or ignored.
"profile" is only recognized in "v2" mode.

 > Running it with awssdk=v2 seems to run into issues with not supporting the sso-session property in the aws config file, or if you switch to the legacy credential configuration for aws, it can't seem to find the cached credentials after you login to aws.

This will almost certainly require fixing in gocloud.dev not pulumi, but do you have any more information on those errors?

**Comment by kg-jcaron** [comment]
Ah actually this is my fault. I was originally on cli version 3.103.0 and encountered the bugs with the v2 query param, and then I was using the v3 query param since it was recommended in the other bug report I linked.

While debugging further I switched to cli 3.105.0 but I guess I didn't try the v2 argument again in the latest version. Just retested it to get the error message for you and it works perfectly now, sorry about that.

Thanks for your help!
